File: 2ndlargest.js
class Solution {
    // Function returns the second largest element
    getSecondLargest(arr) {
        if (arr.length < 2) return -1;

        let largest = -Infinity;
        let secondLargest = -Infinity;

        for (let num of arr) {
            if (num > largest) {
                secondLargest = largest;
                largest = num;
            } else if (num > secondLargest && num !== largest) {
                secondLargest = num;
            }
        }

        return secondLargest === -Infinity ? -1 : secondLargest;
    }
}

// Example usage:
const solution = new Solution();
console.log(solution.getSecondLargest([12, 35, 1, 10, 34, 1])); // Output: 34
console.log(solution.getSecondLargest([10, 5, 10]));            // Output: 5
console.log(solution.getSecondLargest([10, 10, 10]));           // Output: -1

File: python/2ndLargest.py
class Solution:
    # Function returns the second largest element
    def getSecondLargest(self, arr):
        if len(arr) < 2:
            return -1
        
        largest = second_largest = float('-inf')
        
        for num in arr:
            if num > largest:
                second_largest = largest
                largest = num
            elif num > second_largest and num != largest:
                second_largest = num
        
        return -1 if second_largest == float('-inf') else second_largest

# Example usage
solution = Solution()
print(solution.getSecondLargest([12, 35, 1, 10, 34, 1]))  # Output: 34
print(solution.getSecondLargest([10, 5, 10]))            # Output: 5
print(solution.getSecondLargest([10, 10, 10]))           # Output: -1